core/arch/decode_shared.c

```diff
@@ -156,6 +156,22 @@ const char *const size_names[] = {
     "<invalid>" /* was xmm13 */,
     "<invalid>" /* was xmm14 */,
     "<invalid>" /* was xmm15 */,
+    "<invalid>" /* was xmm16 */,
```

Comment:
I think these lines are all incorrect: the OPSZ_ values should not change at all, for backward compatibility.

Reply:
This was needed when adding the DR_REG_XMM15 - DR_REG_XMM31. The assertion 
```
CLIENT_ASSERT(sizeof(size_names) / sizeof(size_names[0]) == OPSZ_LAST_ENUM,
                  "size_names missing an entry");
```
fails. Do I need to add DR_REG_XMM15 - DR_REG_XMM31 below the current DR_REG_YMM15 then? Why is this needed though, what's the compatibility issue here. Do we guarantee the enum values?